tests/resource_tests/calculator_tests/test_ubr_calculator.py

```diff
@@ -0,0 +1,322 @@
+        """
+        # Test with Null and PMI_Skip values
+        # Note: Multiple choice question, pass comma-delimited string.
+        self.assertEqual(self.ubr.ubr_ethnicity(None), UBRValueEnum.NullSkip)
+        self.assertEqual(self.ubr.ubr_ethnicity('PMI_Skip'), UBRValueEnum.NullSkip)
+
+        # Test UBR value
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_AIAN'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_Black'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_MENA'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_Hispanic'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_NHPI'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_RaceEthnicityNoneOfThese'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_White,WhatRaceEthnicity_NHPI'), UBRValueEnum.UBR)
+
+        # Test RBR value
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_White'), UBRValueEnum.RBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('PMI_PreferNotToAnswer'), UBRValueEnum.RBR)
+
+        # Bad or unknown value will default to UBR
+        self.assertEqual(self.ubr.ubr_ethnicity('ABC-123'), UBRValueEnum.UBR)
+
+
+    def test_ubr_geography(self):
+        """
+        UBR Calculator Test - Geography
+        """
+        consent_2014 = datetime(2019, 7, 1).date()  # Hit 2014 version of rural zipcodes.
+        consent_2020 = RURAL_2020_CUTOFF  # Hit 2020 version of rural zipcodes.
+
+        zip_2014 = '35052'  # Only in 2014 file.
+        zip_2020 = '35063'  # Only in 2020 file.
+        zip_both = '72529'  # In both the 2014 file and 2020 file.
+        zip_none = '58493'  # Zip code not in either file.
+
+        # Test with Null and PMI_Skip values
+        self.assertEqual(self.ubr.ubr_geography(consent_2014, None), UBRValueEnum.NullSkip)
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, None), UBRValueEnum.NullSkip)
+
+        # Test zip codes in specific files for UBR
+        self.assertEqual(self.ubr.ubr_geography(consent_2014, zip_2014), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, zip_2020), UBRValueEnum.UBR)
+
+        # Test zip code in both files for UBR
+        self.assertEqual(self.ubr.ubr_geography(consent_2014, zip_both), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, zip_both), UBRValueEnum.UBR)
+
+        # Test for RBR value
+        self.assertEqual(self.ubr.ubr_geography(consent_2014, zip_none), UBRValueEnum.RBR)
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, zip_none), UBRValueEnum.RBR)
+
+        # Bad or unknown value will default to RBR.
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, '123-abc'), UBRValueEnum.RBR)
```

Comment:
Is it worth adding a case with a ZIP+4 response ?

Reply:
Yes, that is a good suggestion, I'll add a couple of test cases.